vm/src/stdlib/array.rs

```diff
@@ -105,23 +106,27 @@ macro_rules! def_array_enum {
             fn count(&self, obj: PyObjectRef, vm: &VirtualMachine) -> PyResult<usize> {
                 match self {
                     $(ArrayContentType::$n(v) => {
-                        Ok(<Option<$t>>::try_from_object(vm, obj)?.map_or(0, |val| {
+                        Ok(if let Ok(val) = $t::try_into_from_object(vm, obj) {
```

Comment:
I don't think this needs to return a result anymore, you can change it to just return `usize`